Index: decision_tree.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+># Decision Tree\r\n\r\nimport csv\r\ndef read_and_write_csv(csv: str) -> None:\r\n    \"\"\"Reads data from our songs dataset\"\"\"\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/decision_tree.py b/decision_tree.py
--- a/decision_tree.py	(revision 655c76930f012979dc522335d7dc1c4938af824d)
+++ b/decision_tree.py	(date 1679949925303)
@@ -1,5 +1,12 @@
 # Decision Tree
 
 import csv
-def read_and_write_csv(csv: str) -> None:
-    """Reads data from our songs dataset"""
+def read_and_write_csv(csv_file: str) -> None:
+    """Reads data from the CSV file that the input refers to and loads it into a new CSV file
+    called songs_final.csv. songs_final.csv will include only the songs and catergories we plan to use.
+
+    Preconditions:
+       - csv_file refers to a valid CSV file in the format described in the project proposal
+    """
+    with open(csv_file) as file:
+        reader = csv.reader(file, delimiter=',')
